internal/types/container/keepalived_container_collector_host.go

```diff
@@ -142,9 +147,42 @@ func (k *KeepalivedContainerCollectorHost) sigNum(sigString string) syscall.Sign
 
 // Signal sends signal to Keepalived process.
 func (k *KeepalivedContainerCollectorHost) signal(signal syscall.Signal) error {
-	err := k.dockerCli.ContainerKill(context.Background(), k.containerName, strconv.Itoa(int(signal)))
+	data, err := os.ReadFile(k.pidPath)
 	if err != nil {
-		logrus.WithError(err).WithField("signal", int(signal)).Error("Failed to send signal")
+		logrus.WithField("path", k.pidPath).WithError(err).Error("Can't find keepalived pid. Send signal to default process")
```

Comment:
Can you change this log to a `Warn`ing instead?

And please change the "Send signal to default process" to  "Falling back to the default process".

Reply:
I believe this needs to be a debug log as this will just generate annoying logs for the ones using signals.
also, I'm wondering why we can't go with something like
```go
if k.pidPath != "" {
  // do exec
} else {
  // do signal
}
```

Reply:
Hi
Because k.pidPath always has the default value "/var/run/keepalived.pid", it can't be judged by `k.pidPath != ""`.
And I think this log is not only a debug log, it can give user some recommendation if the user didn't notice this modification. So I add some more log.
What do you think?

Reply:
Maybe it's better not to have a default value for that then.

Reply:
@TeraTian @clwluvw What do you think guys? Should we remove the default value for `pidPath`?

Reply:
I believe the current implementation will confuse people not intending or intending to use it as it issues the other party anyway without respecting the user's input.

Reply:
I think we should keep this default value. Followings are the questions I ask myself.

**1 Why is the default value set to /var/run/keepalived.pid?**
The exporter isn't meant to run independently; it needs to work alongside keepalived. Since /var/run/keepalived.pid is keepalived’s default PID file location, it makes sense to align with that.

**2 Why have a default value?**
Most users don't want to deal with technical details—they just want the program to work with minimal effort. A default value ensures the program runs correctly without needing manual configuration. If we didn’t have this default, users would have to know the keepalived PID and enter it manually, which adds unnecessary complexity. The simpler, the better.

**3 Does this override user input?**
Not at all. Most users won’t need to change this value unless there’s a specific reason. And if they do, they can easily specify their own value. There’s no restriction on that.

Reply:
Hi @mehdy, any idea about this?

Reply:
Hi @TeraTian,
Thanks. I believe this is good reasoning. It makes sense to keep the default value for `pidPath`.
I'm ok with this setup but with a couple of tweaks.
1. I actually think this should be an INFO-level log. It shouldn't alert the user with `WARN` or `ERROR` and it also should be visible so I wouldn't go with `DEBUG`.
2. I'd suggest removing this part from the log "It's recommended to explicitly specify the pid.". I'm not sure if we should make this recommendation.

Reply:
Yes, I agree with you. I have updated the code. Please review it again, thanks!